client/main.cpp

```diff
@@ -406,6 +411,81 @@ int boinc_main_loop() {
     return finalize();
 }
 
+//
+//
+//
+int connect_sock () {
+
+    int portno = 31416;
+    struct sockaddr_in serveraddr;
+    struct hostent *server;
+    char *hostname = "127.0.0.1";
+    int sockfd;
+
+    server = gethostbyname(hostname);
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    serveraddr.sin_family = AF_INET;
+    serveraddr.sin_port = htons(portno);
+    serveraddr.sin_addr = *((struct in_addr *)server->h_addr);
+    bzero(&(serveraddr.sin_zero), 8);
+
+    int n = connect(sockfd, (struct sockaddr *) &serveraddr, sizeof(struct sockaddr));
+    if (n < 0) {
+       printf("The connect error is : %s", strerror(errno));
```

Comment:
I can't say anything about other code but this line will not work on Windows. You should use GetLastError() instead of errno ( https://msdn.microsoft.com/ru-ru/library/windows/desktop/ms679360%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396 )